internal/models/db/token.go

```diff
@@ -11,6 +11,9 @@ type Token struct {
 	// Unique ID for the token.
 	Id TokenId
 
+	// QueueId of the Queue that this token belongs to
+	QueueId QueueId
+
```

Comment:
We will also need a number field `TokenNumber`:

```
// User facing token number of the queue. Example 1, 2, 3...
TokenNumber uint32
```

You would need to know what's the last issued token number for the queue, and assign it the next number. Can you add it as next PR?

Reply:
Ok
How about adding that to the tokens inside `ReadQueue`?
Cause, the order is also decided inside `ReadQueue` when its sorted by creation time.

Reply:
When you do a ReadToken, you would have to then read from the DB the `count of tokens where creation time < the requested token`.

A token is written once, but read many times. So it's better to be write heavy than read heavy. 

I'll be fine with this as the performance cost is something we can afford to take. You can think a bit more, and take a call in the follow up PR.

Reply:
Added token number